libs/pages/application/src/lib/feature/page-settings-configure-job-feature/page-settings-configure-job-feature.tsx

```diff
@@ -1,193 +1,126 @@
-          } catch (e: unknown) {
-            toastError(e as Error, 'Invalid CMD array')
-            return
-          }
-        } else {
-          schedule.cronjob.arguments = undefined
-        }
-        schedule.cronjob.entrypoint = data.image_entry_point
+    })
+    .with({ job_type: 'LIFECYCLE' }, (s) => {
+      const { on_start, on_delete, on_stop } = s.schedule
+      return {
+        on_start: {
+          enabled: !!on_start,
+          arguments_string:
+            on_start?.arguments && on_start?.arguments?.length > 0 ? JSON.stringify(on_start?.arguments) : undefined,
+          entrypoint: on_start?.entrypoint,
+        },
+        on_stop: {
+          enabled: !!on_stop,
+          arguments_string:
+            on_stop?.arguments && on_stop?.arguments.length > 0 ? JSON.stringify(on_stop?.arguments) : undefined,
+          entrypoint: on_stop?.entrypoint,
+        },
+        on_delete: {
+          enabled: !!on_delete,
+          arguments_string:
+            on_delete?.arguments && on_delete?.arguments.length > 0 ? JSON.stringify(on_delete?.arguments) : undefined,
+          entrypoint: on_delete?.entrypoint,
+        },
       }
-      job.schedule = schedule
-    }
-
-    if (isLifeCycleJob(application)) {
-      const schedule: LifecycleJobResponseAllOfSchedule = {}
-      if (data.on_start?.enabled) {
-        schedule.on_start = {
-          entrypoint: data.on_start.entrypoint,
-          arguments: undefined,
-        }
-
-        if (data.on_start?.arguments_string && data.on_start?.arguments_string.length > 0) {
-          try {
-            schedule.on_start.arguments = eval(data.on_start.arguments_string)
-          } catch (e: unknown) {
-            toastError(e as Error, 'Invalid CMD array')
-            return
-          }
-        }
-      }
-
-      if (data.on_stop?.enabled) {
-        schedule.on_stop = {
-          entrypoint: data.on_stop.entrypoint,
-          arguments: undefined,
-        }
+    })
+    .otherwise(() => undefined)
+
+  const methods = useForm<JobConfigureData>({
+    mode: 'onChange',
+    defaultValues: {
+      max_duration: service?.max_duration_seconds,
+      nb_restarts: service?.max_nb_restart,
+      port: service?.port || undefined,
+      ...schedule,
+    },
+  })
 
-        if (data.on_stop?.arguments_string && data.on_stop?.arguments_string.length > 0) {
-          try {
-            schedule.on_stop.arguments = eval(data.on_stop.arguments_string)
-          } catch (e: unknown) {
-            toastError(e as Error, 'Invalid CMD array')
-            return
+  const onSubmit = methods.handleSubmit((data) => {
+    if (!service) return
+
+    try {
+      const schedule = match(service)
+        .with({ job_type: 'CRON' }, () => {
+          return {
+            cronjob: {
+              scheduled_at: data.schedule || '',
+              entrypoint: data.image_entry_point,
+              arguments: data.cmd_arguments ? eval(data.cmd_arguments) : undefined,
```

Comment:
`eval` is a security issue that should be rewrite
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!

Reply:
Yes I know this point, I can add comment if you want.

I don't rewrite all behavior but if you want to see the history about why we use `eval`, you can check this thread: https://qovery.slack.com/archives/C02NPSG2HBL/p1664352927296669